StreamChatClient_v3/APIClient/Endpoints/Payloads/ChannelEditDetailPayload_Tests.swift

```diff
@@ -0,0 +1,39 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+@testable import StreamChatClient_v3
+import XCTest
+
+class ChannelEditDetailPayload_Tests: XCTestCase {
+    // Test ChannelEditDetailPayload encoded correctly
+    func test_channelEditDetailPayload_encodedCorrectly() throws {
+        let cid = ChannelId.unique
+        let name = "testName"
+        let imageURL = URL(string: "https://getstream.io/random_svg/?id=broken-waterfall-5&amp;name=Broken+waterfall")!
+        let team = "testTeam"
+        let invite: UserId = .unique
+        let extraData: NameAndImageExtraData = .init(name: name, imageURL: imageURL)
+
+        // Create ChannelEditDetailPayload
+        let payload = ChannelEditDetailPayload<DefaultDataTypes>(cid: cid,
+                                                                 team: team,
+                                                                 members: [invite],
+                                                                 invites: [invite],
+                                                                 extraData: extraData)
+
+        let expectedData: [String: Any] = [
+            "name": name,
+            "image": imageURL.absoluteString,
+            "team": team,
+            "members": [invite],
+            "invites": [invite]
+        ]
+
+        let encodedJSON = try JSONEncoder.default.encode(payload)
+        let expectedJSON = try JSONSerialization.data(withJSONObject: expectedData, options: [])
+
+        // Assert ChannelEditDetailPayload encoded correctly
+        AssertJSONEqual(encodedJSON, expectedJSON)
```

Comment:
Haha, we have overloads that either take `Data` and `[String: Any]` but you can't combine them. Maybe it would be worth it, to add such an overload, too? So we don't need to serialize the dictionary every time.

Reply:
Makes sense.

Reply:
Checked it and decided to skip cause turned out it's not that straightforward. When we casting `Data` to `[String: Any]`, `Any` sometimes can be something unexpected. (e.g. Array<String> with one element after encoding and then casting to Any became NSSingleObjectArray and comparison to Array<String> failed)

Reply:
@mantergo ok, noted 👍